utils/tf_solver.py

```diff
@@ -60,7 +101,31 @@ def get_next(stop_val):
 
     def run(self, callback):
         self.model = self.model_init_fn()
-        self.optimizer = self.optimizer_klass(**self.optimizer_kwargs)
+        self.optimizer = self.optimizer_klass(
+            # this scaling is needed as in TF the weight decay is
+            # not multiplied by the learning rate
+            weight_decay=self.decoupled_weight_decay*self.lr,
+            **self.optimizer_kwargs,
```

Comment:
For the learning rate, it seems more classical to give `learning_rate=lr_schedule` no?

Reply:
`lr_schedule` changes the learning rate every iteration and not every epoch.
This is why we need to use the `LearningRateScheduler` API which works with epochs.